### ISSUE TITLE
[+] Create `Files/Dirs/Paths` if they don't exist

### ISSUE THREAD

**Issue by Azathothas** [zer0yu/anew#2]
Hi, your version of anew fails if I try writing to a file that doesn't exist.

```bash
sort -u "this_exists.txt" | anew-rs -q "new_file_that_doesnot_exist.txt"
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

But original anew, works fine, creating files if they don't exist:

```bash
sort -u "this_exists.txt" | anew -q "new_file_that_doesnot_exist.txt"
#works
```

Recommendation: Please implement functions to _auto create the file if it doesn't exist_, so this can be used as **drop-in-replacement for original anew.**

Also, if it is possible, also create directories alongside files, like how `mkdir -p` works. This is not in original anew, but would be immensely useful

**Comment by zer0yu** [comment]
Thank you for bringing this issue to our attention. I'm pleased to inform you that we have released version v0.1.1 of our tool, which addresses the concerns you raised. In this update, we've implemented functionality that ensures the tool can now automatically create both the file and any necessary directories if they do not exist, similar to the behavior of `mkdir -p`. This should make our tool a drop-in replacement for the original anew, with the added benefit of directory creation you suggested.

Please try out the latest version and let us know if it resolves the issue for you. We appreciate your feedback and are always looking for ways to improve.

**Comment by zer0yu** [comment]
I highly recommend trying the `-s` option instead of the `sort -u` command. The relevant usage reference is shown below
```
❯ cat newthings.txt | ./anew things.txt -q -s

❯ cat things.txt
Five
Four
One
Three
Two
Zero
```
